src/ray/raylet/node_manager.cc

```diff
@@ -360,8 +373,28 @@ void NodeManager::ClientAdded(const ClientTableData &client_data) {
   RAY_LOG(DEBUG) << "[ClientAdded] Received callback from client id " << client_id;
   if (client_id == gcs_client_->client_table().GetLocalClientId()) {
     // We got a notification for ourselves, so we are connected to the GCS now.
-    // Save this NodeManager's resource information in the cluster resource map.
-    cluster_resource_map_[client_id] = initial_config_.resource_config;
+
+    // Delay execution to make sure,
+    // 1) Client cache is up to date.
+    // 2) Lookup resource table after subscribe.
+    io_service_.post([this]() {
+      // Fetch resource info for all clients and update cluster resource map.
```

Comment:
Maybe I'm misunderstanding something. 
Why do we need to fetch resource info for all clients here?
This is the callback of one node being added. Why not just fetch that node?

Reply:
It seems like this lambda is called only when the callback is for the local (self) raylet. This is the case when the raylet has been newly instantiated and has no resource state of other raylets. This lambda populates the resources in the raylet by querying the resource table and calling `ResourceCreateUpdated` on the resources of all nodes. Is that correct @kfstorm?

Reply:
I see. thanks.

Reply:
@romilbhardwaj You are right. Previously the resource information is stored in the client table, so a client table notification carries resource update information. Now the two kinds of data are separated, we need some extra logic to make sure local resource information is up to date.

Reply:
I updated the code to lookup resources for only one client at the end of `ClientAdded`.